src/workerd/api/actor-state.c++

```diff
@@ -939,6 +939,45 @@ kj::Maybe<kj::Date> DurableObjectState::getWebSocketAutoResponseTimestamp(jsg::R
   return ws->getAutoResponseTimestamp();
 }
 
+void DurableObjectState::setHibernatableWebSocketEventTimeout(kj::Maybe<uint32_t> timeout) {
+  auto& a = KJ_REQUIRE_NONNULL(IoContext::current().getActor());
+  auto deleteTimeout = true;
+
+  KJ_IF_SOME(t, timeout) {
+    if (t > 0) {
+      // We want to limit the duration of an event to a maximum of 7 days (604800 * 1000 millis).
+      if (t > 604800 * 1000) {
```

Comment:
nit: `JSG_REQUIRE(t <= 604800 * 1000, Error, "Event timeout should not exceed 604800000 ms.")`